src/eval/ops.rs

```diff
@@ -423,3 +425,91 @@ pub fn contains(lhs: &Value, rhs: &Value) -> Option<bool> {
+    let mut piped_args: Vec<Value>;
+    if let Some(lhs) = b.as_untyped().cast_first_match::<ast::PipedSpread>() {
+        let Some(expr) = &lhs.as_untyped().cast_first_match::<Expr>() else {unreachable!()};
+        let expr = expr.eval(vm)?;
+        match expr {
+            Array(arr) => {
+                piped_args = arr.into_iter().collect();
+            }
+            // Dict(dict) => {
+            //     panic!("dicts not supported yet")
+            // } // trickier than it looks like + is it really useful?
+            value => {
+                return Err(eco_format!(
+                    "Can't spread on a single value of type {}, only array are allowed.",
+                    value.type_name()
+                ))
+                .at(span);
+            }
+        };
+    } else {
+        piped_args = vec![b.lhs().eval(vm)?];
+    }
+
+    if let Expr::FuncCall(fc) = rhs {
+        piped_args.reverse();
+        let mut pop_arg = move |span| {
+            if let Some(arg) = piped_args.pop() {
+                Ok(arg)
+            } else {
+                Err(eco_format!("not enough value to pipe",)).at(span)
+            }
+        };
+        // tries to pattern
+        let (callee, syntax_args) = (fc.callee(), fc.args());
+
+        //let args = syntax_args.eval(vm)?;
+        //let mut args = args.items.into_iter().collect::<Vec<_>>();
+        //args.reverse();
+        let mut new_args = crate::eval::Args::new(span, vec![]);
+        for arg in syntax_args.items() {
+            match arg {
+                Arg::Pos(expr) => {
+                    if let Some(ident) = expr.as_untyped().cast::<Ident>() {
+                        if ident.get() == "_" {
+                            new_args.push(span, pop_arg(span)?) // good error instead
+                        } else {
+                            let arg = expr.eval(vm)?;
+                            new_args.push(span, arg)
+                        }
+                    } else {
+                        let arg = expr.eval(vm)?;
+                        new_args.push(span, arg)
+                    };
+                }
+                Arg::Named(n) => {
+                    if_chain::if_chain!(
+                        if let Some(ident) = n.expr().as_untyped().cast::<Ident>();
+                        if ident.get() == "_";
+                        then {
+                            new_args.push_named(span, pop_arg(span)?, &n.name().get());
+                        }
+                        else {
+                            let arg = n.expr().eval(vm)?;
+                            new_args.push_named(span, arg, &n.name().get());
+                        }
+                    );
+                }
+                Arg::Spread(_) => todo!("dont support spread yet"),
+            }
+        }
+        while let Ok(value) = pop_arg(span) {
+            new_args.push(span, value)
+        }
+
+        let func = callee.eval(vm)?.cast::<Func>().unwrap();
```

Comment:
This currently doesn't work with: 
```[Lorem Ipsum] |> text.with(size: 14pt)```
because it tries to evaluate
```rust
[src/eval/ops.rs:506] &callee = FieldAccess(
    FieldAccess(
        FieldAccess: 9 [
            Ident: "text",
            Dot: ".",
            Ident: "with",
        ],
    ),
)
```
which fails with `cannot access fields on type function`.
It would be convenient if `text.with` returned a function instead of failing.
It's coherent with `text` returning a function.
Should I try my hand at it or try find a way to work around it (at least get a satisfaying error)